server/src/main/java/io/spine/server/delivery/CatchUpStation.java

```diff
@@ -0,0 +1,252 @@
+                    if (jobStatus == STARTED) {
+                        if (message.getStatus() == TO_CATCH_UP) {
+                            if (dispatchToCatchUp.containsKey(dispatchingId)) {
+                                conveyor.remove(message);
+                            } else {
+                                dispatchToCatchUp.put(dispatchingId, message);
+                            }
+                        } else if (message.getStatus() == TO_DELIVER) {
+                            conveyor.remove(message);
+                        }
+                    } else if (jobStatus == FINALIZING) {
+                        if (message.getStatus() == TO_DELIVER) {
+                            conveyor.markCatchUp(message);
+                        }
+                    } else if (jobStatus == COMPLETED) {
+                        if (message.getStatus() == TO_CATCH_UP) {
+                            if (!dispatchToCatchUp.containsKey(dispatchingId)) {
+                                dispatchToCatchUp.put(dispatchingId, message);
+                                conveyor.keepForLonger(message, HOW_LONG_TO_KEEP);
+                            } else {
+                                conveyor.remove(message);
+                            }
+                        } else if (message.getStatus() == TO_DELIVER
+                                && dispatchToCatchUp.containsKey(dispatchingId)) {
+                            conveyor.remove(message);
+                        }
+                    }
+                }
+            }
+        }
+
+        return dispatch(conveyor, dispatchToCatchUp);
+    }
+
+    private Result dispatch(Conveyor conveyor, Map<DispatchingId, InboxMessage> dispatchToCatchUp) {
+        if (!dispatchToCatchUp.isEmpty()) {
+            List<InboxMessage> messages = new ArrayList<>(dispatchToCatchUp.values());
+            messages.sort(COMPARATOR);
+
+            DeliveryErrors errors = action.executeFor(messages);
+            conveyor.markDelivered(messages);
+            Result result = new Result(dispatchToCatchUp.size(), errors);
+            return result;
+        }
+        return emptyResult();
+    }
+
+    /**
+     * Tells whether the job matched the passed {@code InboxMessage}.
+     *
+     * <p>To match, two conditions must be met:
+     *
+     * <ol>
+     *     <li>the target entity type of the job and the message must be the same;</li>
+     *
+     *     <li>the identifier of the message target must be included into the list of the
+     *     identifiers specified in the job OR the job matches all the targets of the entity type.
+     *     </li>
+     * </ol>
+     *
+     * @param job
+     *         the catch-up job
+     * @param message
+     *         the message to match to the job
+     * @return {@code true} if the message matches the job, {@code false} otherwise
+     */
+    @VisibleForTesting
+    static boolean matches(CatchUp job, InboxMessage message) {
```

Comment:
This looks like a candidate for a mixin method for `CatchUpMixin` (which I previously commented as probably not used).